### PR TITLE
Add support to query chunk metadata.

### PR DESCRIPTION
**Pull request by avrabe**
For my project I need to access the metadata of each chunk.

Signed-off-by: Ralf Anton Beier <ralf_beier@me.com>

### CODE REVIEW

File: hawkbit/src/ddi/deployment_base.rs [hunk 1]
```diff
@@ -92,7 +93,7 @@ struct ChunkInternal {
     artifacts: Vec<ArtifactInternal>,
 }
 
-#[derive(Debug, Deserialize)]
+#[derive(Clone, Debug, Deserialize)]
```
Comment (gdesmott) [inline hawkbit/src/ddi/deployment_base.rs]: You don't need `Clone` here as you manually clone the key and the value.

File: hawkbit/src/ddi/deployment_base.rs [hunk 2]
```diff
@@ -284,6 +285,15 @@ impl<'a> Chunk<'a> {
             .map(move |a| Artifact::new(a, client.clone()))
     }
 
+    /// An iterator on all the metadata of the chunk.
```
Comment (gdesmott) [inline hawkbit/src/ddi/deployment_base.rs:287]: Doc mentions an iterator but you are actually returning a `HashMap`. I think an iterator would make the more sense here.  Would this fit your needs?

```rust
pub fn metadata(&self) -> impl Iterator<Item = (&str, &str)> {
        self.chunk
            .metadata
            .iter()
            .map(|a| (a.key.as_str(), a.value.as_str()))
    }
```

**Review by gdesmott — CHANGES_REQUESTED** [review]
Thanks for your contribution! 

Can you please run `cargo fmt` on your patch. As you can see, the CI is not passing because of it.

File: hawkbit/src/ddi/deployment_base.rs (hunk 1 above)
Comment (avrabe) [reply hawkbit/src/ddi/deployment_base.rs]: Forgot to remove the unneeded `Clone` from previous tests.
Removed it

File: hawkbit/src/ddi/deployment_base.rs (hunk 2 above)
Comment (avrabe) [reply hawkbit/src/ddi/deployment_base.rs:287]: That also works for me. Changed it.

**Review by gdesmott — APPROVED** [review]

**Comment by gdesmott** [comment]
I just released 0.4.0 with this patch.

**Pull request merged**